proof/crefine/ARM_HYP/Retype_C.thy

```diff
@@ -7990,6 +8184,14 @@ lemma insertNewCap_ccorres:
   apply (simp add: untypedZeroRange_def Let_def)
   done
 
+lemma Arch_createObject_not_untyped:
+  "\<forall>s. \<Gamma>\<turnstile>\<^bsub>/UNIV\<^esub>
+   {s} Call Arch_createObject_'proc {t. cap_get_tag (ret__struct_cap_C_' t) \<noteq> scast cap_untyped_cap}"
```

Comment:
I'm a bit lost as to why this spec override is needed (i.e. how you came up with it)

Reply:
The proof that follows this one previously just ran`vcg` which happened to either unfold everything or use `modifies` rules for `Arch_createObject`, depending on which arch it was from. That unfolding now draws in cache functions which have loops and are quite big.

It turns out, that the proof doesn't actually need anything at all from `Arch_createObject`: it was only interested in the case where a new untyped cap was returned, which `Arch_createObject` never does. So I proved that instead, which can also be done without further unfolding inside `Arch_createObject` and leads to faster proofs all around.

Reply:
Nice. This is a super-annoying part of CRefine, making sure to always keep one's eye out for situations like this, preventing explosions.